### PR TITLE
Gutter fun

### PR DESCRIPTION
**Pull request by rgbkrk**
This introduces a semi-modular status bar, styled akin to the Atom status bar.

screenshot 2015-06-29 17 03 01

I don't actually mean to have text like the above, but I wanted to flesh out this custom component to accept various statuses (as well as kernel icons).

### CODE REVIEW

**Comment by rgbkrk** [comment]
The next thing I need to do here is improve our message dispatch. It's tied too closely in with the display.

**Comment by rgbkrk** [comment]
screenshot 2015-06-29 17 03 01

**Comment by rgbkrk** [comment]
status

Now with updating status bar.

**Comment by rgbkrk** [comment]
Am I crazy or does sidecar render "Kernel: busy" faster than my console moves to the next line?

**Comment by rgbkrk** [comment]
The "API" for the gutter is now:

```
<sidecar-gutter status="idle">
```

Just set the status and off you go!

Next up is handling the kernel type. :wink:

**Comment by rgbkrk** [comment]
I'm tempted to see when declaring functions on the prototypes of custom elements would be useful.

E.g. if we have an element `gutter = document.querySelector("sidecar-gutter")` and `foobar` is defined on it, we can then run that function directly `gutter.foobar('great');`.

**Comment by rgbkrk** [comment]
Now that I have it working, it would be good to stand back and think both about API design for this element as well as design of the element on the page.

**Comment by jdfreder** [comment]
> Am I crazy or does sidecar render "Kernel: busy" faster than my console moves to the next line?

You are not crazy, I can confirm this.

File: templates/sidecar-gutter.html [hunk 1]
```diff
@@ -16,11 +16,53 @@
...
+      white-space: nowrap;
+      overflow: hidden;
+      cursor: default;
+      -webkit-user-select: none;
+
+
+      min-width: -webkit-min-content;
+    }
+
+    .flexbox-repaint-hack {
```
Comment (jdfreder) [inline templates/sidecar-gutter.html]: Why is this called repaint hack?  What does it actually do?

Comment (rgbkrk) [inline templates/sidecar-gutter.html]: I completely stole the "repaint hack", name and everything from Atom's status bar.

**Comment by rgbkrk** [comment]
Final state of the night:

screenshot 2015-06-29 23 15 15

**Comment by rgbkrk** [comment]
Nevermind, decided to use a box shadow. The shadow on top of the status bar might be a bad idea though. :unamused:

**Comment by rgbkrk** [comment]
screenshot 2015-06-29 23 21 31

**Comment by jdfreder** [comment]
I like it with the shadow

**Comment by rgbkrk** [comment]
> I like it with the shadow

Alright, I'mma gonna merge it then. :wink:

**Comment by smashwilson** [comment]
@rgbkrk Yeah, that looks awesome :grin:

I liked the "Now you're cooking with sidecar" message. We should do a random message-of-the-day there while it's idle :wink:

**Comment by rgbkrk** [comment]
I seriously considered making the `<content>` area reserved for a message of the day. Nothing wrong with adding it back in though.

**Pull request merged**